pkg/client/smd.go

```diff
@@ -151,7 +151,7 @@ func (client *SmdClient) makeRequest(endpoint string) ([]byte, error) {
 	}
 
 	// fetch DHCP related information from SMD's endpoint:
-	url := fmt.Sprintf("%s:%d/hsm/v2%s", client.Host, client.Port, endpoint)
+	url := fmt.Sprintf("%s/hsm/v2%s", client.Host, endpoint)
```

Comment:
I think this is fine for now, but I'm getting:
```json
{"level":"error","time":1733940654,"message":"failed to generate file: failed to fetch ethernet interfaces with client: failed to read HTTP response: failed to make request: Get \"/hsm/v2/Inventory/EthernetInterfaces\": unsupported protocol scheme \"\""}
```

even when the host is defined.

I think we need to pass all of the `ClientOption`s to the `SmdClient` when creating here:

https://github.com/OpenCHAMI/configurator/blob/5b351d2cac2476601e53ffa4eace12e34a9bdeb0/pkg/generator/generator.go#L390

Reply:
Is this the right branch? I don't see a line 390 in the PR branch.

Reply:
Argh... GitHub switched back to `main` when I clicked on the symbol. Here is the line on this branch:

https://github.com/OpenCHAMI/configurator/blob/ebe4e02cf0f670ebc1b6cadc5bf606950719cb7f/pkg/server/server.go#L55

Reply:
Which plugin are you trying to use here? I think it's probably worth to include more information in the error.

Reply:
I'm running the server as follows (config displayed):
```
# ./configurator serve --verbose --cacert /opt/shared/ochami.pem --config ./config.yaml
{
        "Version": "",
        "Server": {
                "Host": "127.0.0.1:3334",
                "Port": 0,
                "Jwks": {
                        "Uri": "",
                        "Retries": 5
                }
        },
        "SmdClient": {
                "Host": "http://127.0.0.1:27779",
                "Port": 0,
                "AccessToken": ""
        },
        "AccessToken": "eyJhb...",
        "Targets": {
                "dnsmasq": {
                        "Plugin": "",
                        "TemplatePaths": null,
                        "FilePaths": null,
                        "RunTargets": null
                }
        },
        "PluginDirs": [],
        "CertPath": ""
}
{"level":"error","time":1734034071,"message":"failed to generate file: failed to fetch ethernet interfaces with client: failed to read HTTP response: failed to make request: Get \"/hsm/v2/Inventory/EthernetInterfaces\": unsupported protocol scheme \"\""}                   
1:07PM INF Request bytes_in=0 bytes_out=203 duration=0.115345 method=GET remote_addr=127.0.0.1:44354 request_id=ochami-vm/WwwPpCGpkq-000001 request_uri=/generate?target=dnsmasq status="Internal Server Error" status_code=500 user_agent=configurator                          
2024/12/12 13:07:51 [ochami-vm/WwwPpCGpkq-000001] "GET http://127.0.0.1:3334/generate?target=dnsmasq HTTP/1.1" from 127.0.0.1:44354 - 500 203B in 214.879µs 
```
The response above is printed when I run:
```
./configurator fetch --target dnsmasq --host http://127.0.0.1:3334
```
I believe the line I linked is related to this.

Reply:
Actually, I think the problem is here:
https://github.com/OpenCHAMI/configurator/blob/cccf6321cc9f37de8609d5bc9632f1a171d84e7b/pkg/server/server.go#L92-L95
None of the `SmdClient` options are getting read in the `serve` command, and so the host field is blank when making the request.

Reply:
I think the issue may actually be because of this:
https://github.com/OpenCHAMI/configurator/blob/readme-and-improvements/cmd/serve.go#L45

Reply:
So I have been tracking down the issue by print statements, etc. and I that, when passing `--target dnsmasq` to the `fetch` command, the output of `s.getTarget` is nil here:
https://github.com/OpenCHAMI/configurator/blob/cccf6321cc9f37de8609d5bc9632f1a171d84e7b/pkg/server/server.go#L142
and so `GenerateWithTarget` is called here:
https://github.com/OpenCHAMI/configurator/blob/cccf6321cc9f37de8609d5bc9632f1a171d84e7b/pkg/server/server.go#L160
instead of `Generate` here:
https://github.com/OpenCHAMI/configurator/blob/cccf6321cc9f37de8609d5bc9632f1a171d84e7b/pkg/server/server.go#L154
which would pass the `s.GeneratorParams` containing the SMD host configuration.

It looks like `s.getTarget` returns nil when the target (`dnsmasq` in this case) doesn't exist in the server's `Targets` map. Does this get set somewhere? The `server.New()` function doesn't seem to set this.

Reply:
The new commit (pending review comment) seems to fix this, however the configurator doesn't seem to contact SMD despite being configured.

```
# ./configurator serve --verbose --cacert /opt/shared/ochami.pem --config ./config.yaml
{"level":"warn","time":"2024-12-17T22:27:27-07:00","message":"No token found. Continuing without one...\n"}
{                                   
        "Version": "",
        "Server": {       
                "Host": "127.0.0.1:3334",
                "Port": 0,                            
                "Jwks": {
                        "Uri": "",          
                        "Retries": 5
                }                          
        },                      
        "SmdClient": {
                "Host": "http://127.0.0.1:27779",      
                "Port": 0,                                      
                "AccessToken": ""                        
        },                                                                    
        "AccessToken": "",                                                                                                                                                                                  
        "Targets": {                                                     
                "dnsmasq": {                                                  
                        "Plugin": "",                                        
                        "TemplatePaths": null,                               
                        "FilePaths": null,                                     
                        "RunTargets": null                                     
                }                                                                   
        },                                                                       
        "PluginDirs": [],                                                        
        "CertPath": ""                                                           
}                                                          
{"level":"error","time":1734499653,"message":"could not find target in default generators"}
10:27PM INF Request bytes_in=0 bytes_out=2 duration=0.076701 method=GET remote_addr=127.0.0.1:39508 request_id=te-head.si.usrc/XNIz1WqTBL-000001 request_uri=/generate?target=dnsmasq status=OK status_code=2
00 user_agent=configurator                
2024/12/17 22:27:33 [te-head.si.usrc/XNIz1WqTBL-000001] "GET http://127.0.0.1:3334/generate?target=dnsmasq HTTP/1.1" from 127.0.0.1:39508 - 200 2B in 149.241µs
{"level":"error","time":1734499667,"message":"could not find target in default generators"}
```
Using the following client command:
```
./configurator fetch --target dnsmasq --host http://127.0.0.1:3334 --verbose
```